homeassistant/components/envisalink/alarm_control_panel.py

```diff
@@ -44,6 +52,21 @@
 
     async_add_entities(devices)
 
+    @callback
+    def alarm_arm_max_handler(service):
+        """Map arm max service to method on the Alarm."""
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        code = service.data.get(ATTR_CODE)
+        target_devices = [device for device in devices
+                          if device.entity_id in entity_ids]
+
+        for device in target_devices:
+            device.async_alarm_arm_max(code)
+
+    hass.services.async_register(
+        alarm.DOMAIN, SERVICE_ALARM_ARM_MAX, alarm_arm_max_handler,
```

Comment:
Custom services should be registered under the integrating component domain, ie envisalink. That's our new way of structuring things.